libvast/src/system/indexer_manager.cpp

```diff
@@ -0,0 +1,102 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/system/indexer_manager.hpp"
+
+#include <caf/exit_reason.hpp>
+#include <caf/local_actor.hpp>
+#include <caf/send.hpp>
+
+#include "vast/load.hpp"
+#include "vast/save.hpp"
+#include "vast/system/indexer.hpp"
+
+namespace vast::system {
+
+indexer_manager::indexer_manager(path dir, indexer_factory f)
+  : dir_(dir),
+    make_event_indexer_(std::move(f)) {
+  // If the directory exists already, we must have some state from the past
+  // and are pre-loading all INDEXER types we are aware of.
+  if (exists(dir_)) {
+    if (auto result = load(dir_ / "meta", meta_data_)) {
+      for (auto& [str, t] : meta_data_.types)
+        indexers_.emplace(t, make_event_indexer(t));
+    } else {
+      VAST_ERROR("unable to read INDEXER types:", result.error());
+    }
+  }
+}
+
+indexer_manager::~indexer_manager() noexcept{
```

Comment:
Is it okay to have a `noexcept` label in this function, which performs I/O and message passing?

Reply:
Destructors are always `noexcept` by default. Throwing in a destructor is a very bad idea anyways and hitting `terminate` in this case is the best thing to happen in most cases.